### PR TITLE
docs: implement "Dataset Management" in chapter "FEATURES"

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2bgyqwd SDK 文档编写 —— Features > Dataset Management](https://app.clickup.com/t/2bgyqwd)

File: docs/source/features/dataset_management.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+.. code:: python
+
+   from graviti import Workspace
+   ws = Workspace(f"{YOUR_ACCESSKEY}")
+
+******************
+ Create a Dataset
+******************
+
+SDK provide function :func:`~graviti.manager.dataset.DatasetManager.create` to support creating
```
Comment (linjiX) [inline docs/source/features/dataset_management.rst:21]: `provide` or `provides`

Comment (linjiX) [reply docs/source/features/dataset_management.rst:21]: normally we use the word `method` to represent the function in class.

**Review by linjiX — CHANGES_REQUESTED** [review]

File: docs/source/features/dataset_management.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                       "value": "daytime"
+                   }
+               ]
+           }
+       ]
+   }
+   dataset.search("train", criteria)
+
+.. note::
+   The search criteria consist of three parts:
```
Comment (Lee-000) [inline docs/source/features/dataset_management.rst:119]: `consists`?

File: docs/source/features/dataset_management.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+   * key: The column name of the DataFrame. Multiple indexes can be expressed using ``.``.
```
Comment (Lee-000) [inline docs/source/features/dataset_management.rst:121]: Would `connected with` or `joined by` be better than `expressed using`?

File: docs/source/features/dataset_management.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+   * value: The value corresponding to the operator.
+   * | opt: The operational relationship between the key and value, or the logical relationship
```
Comment (Lee-000) [inline docs/source/features/dataset_management.rst:123]: Should the order be consistent with the previous example?

File: docs/source/features/dataset_management.rst (hunk 2 above)
Comment (graczhual) [reply docs/source/features/dataset_management.rst:119]: no, this is the plural form.

**Review by AChenQ — APPROVED** [review]

File: docs/source/features/dataset_management.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+.. code:: python
+
+   from graviti import Workspace
+   ws = Workspace(f"{YOUR_ACCESSKEY}")
+
+******************
+ Create a Dataset
+******************
+
+SDK provides method :func:`~graviti.manager.dataset.DatasetManager.create` to support creating
```
Comment (linjiX) [inline docs/source/features/dataset_management.rst:21]: what is the difference between sphinx `:func:` and `:meth:`

Comment (graczhual) [reply docs/source/features/dataset_management.rst:21]: no difference here
https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#python-roles
https://stackoverflow.com/questions/34848253/whats-the-difference-in-behaviour-between-func-and-meth-roles-in-python-sph

Comment (linjiX) [reply docs/source/features/dataset_management.rst:21]: will it be better if we use `:meth:` here?

Comment (graczhual) [reply docs/source/features/dataset_management.rst:21]: OK

**Review by linjiX — APPROVED** [review]

**Pull request merged**